server/script/migrateAssetsLocation.js

```diff
@@ -0,0 +1,252 @@
+  });
+  return Promise.each(revisions, async it => {
+    const payload = await migrateRevision(it);
+    return it.update(payload, { transaction });
+  });
+}
+
+async function migrateActivity(activity) {
+  const { repositoryId, data: meta } = activity;
+  const fileMetas = getActivityMetadata(activity)
+    .filter(it => it.type === 'FILE')
+    .map(it => it.key);
+  const data = await getNewMeta(fileMetas, meta, repositoryId);
+  return { data };
+}
+
+async function migrateContentElement(element) {
+  const data = await migrateContentElementData(element);
+  const meta = await migrateContentElementMeta(element);
+  return { data, meta };
+}
+
+function migrateContentElementData(element) {
+  const { type, data } = element;
+  if (type === 'IMAGE') return imageMigrationHandler(element);
+  if (data.embeds) return embedsMigrationHandler(element);
+  if (data.assets) return defaultMigrationHandler(element);
+  return data;
+}
+
+async function migrateContentElementMeta(element) {
+  const { repositoryId } = element;
+  const metaInputs = schemasIds
+    .map(id => getElementMetadata(id, element))
+    .filter(meta => !meta.isEmpty)
+    .map(meta => meta.inputs);
+  const fileMetas = flatten(metaInputs)
+    .filter(it => it.type === 'FILE')
+    .map(it => it.key);
+  return getNewMeta(fileMetas, element.meta, repositoryId);
+}
+
+async function getNewMeta(fileMetas, meta, repositoryId) {
+  const newMeta = await Promise.map(toPairs(meta), async it => {
+    const [id, value] = it;
+    if (!fileMetas.includes(id)) return it;
+    const url = get(value, 'url');
+    if (!url) return it;
+    const { key, newKey } = getKeysFromUrl(url, repositoryId) || {};
+    if (!key || !newKey) return it;
+    await storage.copyFile(key, newKey);
+    return [id, {
+      ...value,
+      key: newKey,
+      url: `${protocol}${newKey}`,
+      publicUrl: await storage.getFileUrl(newKey)
+    }];
+  });
+  return fromPairs(newMeta);
+}
+
+async function imageMigrationHandler(element) {
+  const { repositoryId, data } = element;
+  const url = get(data, 'url');
+  if (!url) return data;
+  const { key, newKey } = getKeysFromUrl(url, repositoryId) || {};
+  if (!key || !newKey) return data;
+  await storage.copyFile(key, newKey);
+  return { ...element.data, url: newKey };
+}
+
+async function embedsMigrationHandler(element) {
+  const { repositoryId, data } = element;
+  const embeds = await getMigratedEmbeds(repositoryId, data.embeds);
+  return { ...data, embeds };
+}
+
+function getMigratedEmbeds(repositoryId, embeds) {
+  return Promise.map(
+    Object.entries(embeds),
+    async ([id, embed]) => {
+      const payload = await migrateContentElement({ repositoryId, ...embed });
+      return [id, { ...embed, ...payload }];
+    }
+  ).reduce((acc, [id, embed]) => ({ ...acc, [id]: embed }), {});
+}
+
+async function defaultMigrationHandler(element) {
+  const { repositoryId, data } = element;
+  const url = get(element, 'data.assets.url');
```

Comment:
Why picking up .url 🤔 Any key within assets dir that has storage prefix should be resolved ⚠️